.github/workflows/checks.yml

```diff
@@ -97,53 +95,43 @@ jobs:
           poetry install
 
       # ====================== CONTRACTS v2 ====================== #
-
       - name: Cache contracts v2
-        id: cache-contracts-v2
+        id: cache-contracts_v2
         uses: actions/cache@v3
         with:
-          path: starknet_py/tests/e2e/mock/contracts_compiled_v2
-          key: ${{ runner.os }}-contracts-${{ hashFiles('starknet_py/tests/e2e/mock/contracts_v2') }}-${{ env.CAIRO_V2_SHA }}
-    
-      - uses: actions/checkout@v4
-        with:
-          repository: starkware-libs/cairo
-          path: cairo_v2
-          ref: ${{ env.CAIRO_V2_SHA }}
+          path: starknet_py/tests/e2e/mock/contracts_v2/target
+          key: ${{ runner.os }}-contracts-${{ hashFiles('starknet_py/tests/e2e/mock/contracts_v2') }}
 
-      - name: Create text file
-        run: |
-          echo "${{ github.workspace }}/cairo_v2/Cargo.toml" > ${{ github.workspace }}/starknet_py/tests/e2e/manifest-path
 
-      - name: Compile contracts
-        if: steps.cache-contracts-v2.outputs.cache-hit != 'true'
+      - uses: software-mansion/setup-scarb@v1.3.2
+        with:
+          tool-versions: "starknet_py/tests/e2e/mock/contracts_v2/.tool-versions"
+
+      - name: Compile contracts v2
+        working-directory: "starknet_py/tests/e2e/mock/contracts_v2"
+        if: steps.cache-contracts_v2.outputs.cache-hit != 'true'
         run: |
-          poetry run poe compile_contracts_v2
+          scarb build
```

Comment:
it may be a good idea to use our new script to build contracts on CI as well

Reply:
It works!